gflownet/envs/cube.py

```diff
@@ -0,0 +1,1433 @@
+"""
+Classes to represent hyper-cube environments
+"""
+import itertools
+import warnings
+from abc import ABC, abstractmethod
+from typing import List, Optional, Tuple
+
+import matplotlib.pyplot as plt
+import numpy as np
+import torch
+from sklearn.neighbors import KernelDensity
+from torch.distributions import Bernoulli, Beta, Categorical, MixtureSameFamily
+from torchtyping import TensorType
+
+from gflownet.envs.base import GFlowNetEnv
+from gflownet.utils.common import copy, tbool, tfloat
+
+
+class CubeBase(GFlowNetEnv, ABC):
```

Comment:
As a general rule I would advise against creating classes like this before they are actually required (subclassed by more than one class) - though I am not suggesting removing it at this point.

Reply:
I agree. In my defense, when I created it there where 2 subclasses. Then I ended up removing one for now.